CIP-0XXX/README.md

```diff
@@ -0,0 +1,297 @@
+
+Currently, certification of the Cardano node database relies on the internal representation of the historical Cardano node database:
+
+- Immutable files (chunk files and primary/secondary indexes) are deterministically computed by the historical Cardano node. They are signed and served by Mithril.
+- Ledger state snapshots, however, are not consistently computed at the same blockchain point across nodes. They are signed using IOG-owned keys instead and they are served by Mithril.
+
+This limitation means only the historical Cardano node fully benefits from ledger state and immutable file snapshots, and the ledger state snapshot is not yet a first-class citizen in the Mithril protocol.
+
+Another use case for the ledger state snapshot format is to facilitate debugging and testing of a node implementation.
+
+## Specification
+
+<!-- The technical specification should describe the proposed improvement in sufficient technical detail. In particular, it should provide enough information that an implementation can be performed solely on the basis of the design in the CIP. This is necessary to facilitate multiple, interoperable implementations. This must include how the CIP should be versioned, if not covered under an optional Versioning main heading. If a proposal defines structure of on-chain data it must include a CDDL schema in its specification.-->
+
+### Ledger state snapshot format
+
+#### Requirements
+
+The ledger state snapshot format should:
+
+- Be deterministically encoded.
+- Support both random access and sequential reading.
+- Be adaptable to future changes in the Cardano ecosystem.
+- Ensure forward and backward compatibility, allowing new implementations to read older versions and vice versa.
+- Be defined using a robust schema notation (e.g., CDDL) to guarantee clarity and validation.
+
+We propose deterministically encoded CBOR ([RFC8949 Sec 4.2](https://datatracker.ietf.org/doc/html/rfc8949#section-4.2), with the tighter profile defined in [dCBOR](https://datatracker.ietf.org/doc/draft-mcnally-deterministic-cbor/)) with CDDL schemas as the base format and tooling.
+
+#### Contents
+
+> [!NOTE]
+>
+> - Do we want to include derived data in the snapshot?
+> - Do we want to create a file per type of data or a single file containing all the data?
+
+The essential required data in a snapshot include are:
+
+- UTXOs
+- Stake delegation
+- Reward accounts
+- Protocol parameters
+- Accounting pots (reserves, deposits etc.)
+- SPO state
+- DRep state
+- Governance action general state
+- Governance action voting state
+- Header state (e.g. nonces)
+
+It could maybe also be useful to produce derived data that is commonly useful:
+
+- Payment address balances
```

Comment:
While I think it is interesting to include representations of the data that is commonly produced from the ledger state, I'm not entirely sure it makes sense to include these in a "Canonical Ledger State" specification.

Simplicity is our friend and, while I can appreciate that this is an attempt at simplifying derived data, I would argue it's actually increasing complexity for creating and parsing ledger state which is more important.

I think we should focus on the canonical ledger state being the minimum required data to fully encode ledger state.